### PR TITLE
move all management of jobs executions to ScheduledJobManager

### PR DESCRIPTION
**Pull request by KazankovMarch**
I wanted to resolve issue #71, but management of jobs executions is separated between Worker and ScheduledJobManager

Worker is too complex class, so let's make that only ScheduledJobManager now responsible for management of jobs executions. Such a first step for refactoring Worker class

Note: I didn't change any DJM logic in this PR, I only moved some blocks of code from Worker to ScheduledJobManager; I suggest refactor ScheduledJobManager in next PR

### CODE REVIEW

**Comment by KazankovMarch** [comment]
I forgot to close PersistentExpiringLockManager

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java [hunk 1]
```diff
@@ -20,9 +34,53 @@
...
     //Jobs that currently scheduled. Some of them are running and some will be launched any time.
     private final Map<JobDescriptor, List<ScheduledJobExecution>> scheduledJobs = new ConcurrentHashMap<>();
 
+    public ScheduledJobManager(
+            CuratorFramework curatorFramework,
+            ZkPathsManager paths,
+            Profiler profiler,
+            String workerId,
+            DynamicProperty<DistributedJobManagerSettings> settings,
+            AtomicBoolean isWorkerShutdown
```
Comment (swarmshine) [inline distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java:64]: isWorkerShutdown breaked borders and leaked into ScheduledJobManager. Maybe isWorkerShutdown should stay in worker class, and worker should use and decide whether to call restartJobsAccordingToNewAssignments or not depending on Worker state?

**Review by swarmshine — CHANGES_REQUESTED** [review]

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java [hunk 2]
```diff
@@ -20,9 +34,53 @@
...
+        this.lockManager = new PersistentExpiringLockManager(
+                curatorFramework,
+                settings.map(DistributedJobManagerSettings::getLockManagerConfig),
+                profiler
+        );
+    }
+
     synchronized List<ScheduledJobExecution> getScheduledJobExecutions(JobDescriptor distributedJob) {
         return scheduledJobs.get(distributedJob);
     }
```
[the remark this answers is not in the document]
Comment (KazankovMarch) [reply distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java:86]: fixed

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java (hunk 1 above)
Comment (KazankovMarch) [reply distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java:64]: I agree. I have a suggestion for `ReschedulableScheduler` API: https://github.com/ru-fix/jfix-stdlib/pull/73

**Review by phoenigm — APPROVED** [review]

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java [hunk 3]
```diff
@@ -20,18 +34,62 @@
...
+                threadPoolSize,
+                profiler);
+        this.lockManager = new PersistentExpiringLockManager(
+                curatorFramework,
+                settings.map(DistributedJobManagerSettings::getLockManagerConfig),
+                profiler
+        );
+    }
+
     synchronized List<ScheduledJobExecution> getScheduledJobExecutions(JobDescriptor distributedJob) {
```
Comment (swarmshine) [inline distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java:84]: Probably should be private?

**Review by swarmshine — CHANGES_REQUESTED** [review]

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java (hunk 3 above)
Comment (KazankovMarch) [reply distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java]: fixed

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java (hunk 1 above)
Comment (KazankovMarch) [reply distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java]: fixed

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java [hunk 4]
```diff
@@ -20,18 +33,57 @@
...
+                profiler
+        );
+    }
+
+    private synchronized List<ScheduledJobExecution> getScheduledJobExecutions(JobDescriptor distributedJob) {
         return scheduledJobs.get(distributedJob);
     }
 
-    synchronized void removeIf(Predicate<Map.Entry<JobDescriptor, List<ScheduledJobExecution>>> predicate) {
+    private synchronized void removeIf(Predicate<Map.Entry<JobDescriptor, List<ScheduledJobExecution>>> predicate) {
```
Comment (swarmshine) [inline distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java:82]: Seems like there is no need for `synchronized` for this private methods

**Review by swarmshine — CHANGES_REQUESTED** [review]

**Pull request merged**